app/controllers/inbox/meetings/new.js

```diff
@@ -0,0 +1,42 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { dropTask } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
+
+export default class InboxMeetingsNewController extends Controller {
+  @service router;
+
+  get meeting() {
+    return this.model.meeting;
+  }
+
+  @dropTask
+  * saveMeetingTask(event) {
+    event.preventDefault();
+
+    let bestuursorgaan = yield this.meeting.bestuursorgaan;
+    if (!bestuursorgaan) {
+      this.meeting.errors.add('bestuursorgaan', 'inbox.meetings.new.meeting.errors.administrativeBody.required');
+    }
+
+    if (this.meeting.isValid) {
+      yield this.meeting.save();
+      this.router.replaceWith('meetings.edit', this.meeting.id);
+    }
+  }
+
+  @action
+  updateAdministrativeBody(administrativeBody) {
+    this.meeting.bestuursorgaan = administrativeBody;
+    this.meeting.errors.remove('bestuursorgaan');
+  }
+
+  @action
+  cancelMeetingCreation() {
+    // Prevent unwanted calls due to focus trap...
+    if (this.meeting.isNew && !this.meeting.isDeleted) {
+      this.meeting.destroyRecord();
+      this.router.replaceWith('inbox.meetings');
```

Comment:
1 thing I noticed is that the loading template is shown very briefly when returning back to the `inbox.meetings` route. It looks a bit glitchy / weird, but it's related to the `refreshModel` option of the query params so I don't think I can fix it myself. 

AFAIK the model hooks shouldn't be rerun when transitioning between sibling routes (.new to .index), and the query params don't change either, so I'm not sure if that's a router bug or not.

Reply:
Someone already created an issue for it: https://github.com/emberjs/ember.js/issues/19497